Campus-iOS/Map/MainView/MapViewModel.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MapViewModel.swift
+//  Campus-iOS
+//
+//  Created by Philipp Wenner on 08.03.22.
+//
+
+import Foundation
+import SwiftUI
+import Alamofire
+
+final class MapViewModel: ObservableObject {
+    @Published var canteens: [Cafeteria] = []
+    @Published var selectedCanteen: Cafeteria?
+    @Published var zoomOnUser: Bool = true
+    @Published var panelPosition: String = "pushMid"
+    @Published var selectedCanteenName: String = " "
+    @Published var selectedAnnotationIndex: Int = 0
+    
+    private let endpoint = EatAPI.canteens
+    private let sessionManager = Session.defaultSession
+    
+    func fetchCanteens() {
+        sessionManager.request(endpoint).responseDecodable(of: [Cafeteria].self, decoder: JSONDecoder()) { [self] response in
```

Comment:
Maybe also create a separate `Service` for this network request, as can be seen in the Grade screens